sympy/solvers/ode/systems.py

```diff
@@ -1064,3 +1066,153 @@ def neq_nth_linear_constant_coeff_match(eqs, funcs, t):
         return match
 
     return None
+
+
+def _preprocess_eqs(eqs):
+
+    processed_eqs = []
+    for eq in eqs:
+        processed_eqs.append(eq if isinstance(eq, Equality) else Eq(eq, 0))
+
+    return processed_eqs
+
+
+# For now, this function returns a simple output, later it will
+# be capable of dividing the system of ODEs into strongly and
+# weakly connected components. Also, later the argument funcs
+# is needed too.
+def _component_division(eqs, funcs):
+    return [[[eqs, funcs]]]
+
+
+# Returns: List of equations
+def _linear_ode_solver(match):
+    t = match['t']
+    funcs = match['func']
+
+    rhs = match.get('rhs', None)
+    A = -match['func_coeff']
+    B = match.get('commutative_antiderivative', None)
+    const_idx = match['const_idx']
+    type_of_equation = match['type_of_equation']
+
+    sol_vector = linodesolve(A, t, b=rhs, B=B,
+                             type=type_of_equation, const_idx=const_idx)
+
+    sol = [Eq(f, s) for f, s in zip(funcs, sol_vector)]
+
+    return sol
+
+
+# Returns: (List of List of equations, const_idx) or None
+def _ode_component_solver(eqs, funcs, t, const_idx=0):
+    match = neq_nth_linear_constant_coeff_match(eqs, funcs, t)
+    match['const_idx'] = const_idx
+    match['t'] = t
+
+    if match.get('is_linear', False):
+        return [_linear_ode_solver(match)], const_idx + len(eqs)
+    if match.get('is_implicit', False):
+        canon_eqs = match['canon_eqs']
+        sols = []
+        for canon_eq in canon_eqs:
+            sol, const_idx = _ode_component_solver(canon_eq, funcs, t, const_idx=const_idx)
+            if sol is not None:
+                sols += sol
+        return sols, const_idx
+
+    # To add non-linear case here in future
+
+    return None
+
+
+def dsolve_system(eqs, funcs=None, t=None):
```

Comment:
I think it's better to require that the user pass all arguments like `funcs` and `t`. A user who doesn't want to supply those can call `dsolve`.

Reply:
As I said in the previous comment, in order to do that, it would make calling `dsolve_system` from `dsolve` very inefficient.